### ISSUE TITLE
From time to time Mypal hangs and can be restored only after system reboot (WinXP, 32 bit)

### ISSUE THREAD

**Issue by etsenberg** [Feodor2/Mypal68#477]
It happens after days or weeks of normal work w/o rebooting. I don't know the specific circumstances. Just switching to another task, then switching back - and it hangs. Killing and restarting does not help (the task starts and remains in memory, but nothing is displayed in its window), hibernation and restoring does not help - only rebooting the system. It happened in previous versions and happens in 68.14. When it happens, ALL Mypal versions stop working the same way - none can run until reboot, even if they use different profiles. After rebooting all works again w/o additional actions, so it's not in the profile - it must be something in the memory which does not disappear even when all mypal.exe in the process manager are killed. All other tasks work normally. There is a lot of free memory.
Right now I have this situation. What can I do to diagnose the issue? I will not reboot, waiting for your answer.

**Comment by JimMP68User** [comment]
I have not seen this, and, I surf a lot. Sometimes Mypal (14.2b at present) will lag or stop, not respond for a bit but it always comes back for me, except one occasion where trying to shut it down and something hung up, and I had to use Kill Task in Task Manager.

 Can you give type of computer, memory installed and video card type?

**Comment by etsenberg** [comment]
> Can you give type of computer, memory installed and video card type?

[DxDiag.txt](https://github.com/user-attachments/files/16022651/DxDiag.txt)
See dxdiags.
Hardware acceleration turned off.  
I also checked events and found lots of mypal errors of the same type:

Event Type: Error
Event Source: Application Error
Event Category: None
Event code: 1000
Date: 6/27/2024
Time: 2:22:07 PM
Description:
Application error mypal.exe, version 68.12.0.8721, module xul.dll, version 68.12.0.8721, 

and the same error 1000 in mozglue.dll,. But when they happened on previous days, they did not hinder Mypal to run,

**Comment by Feodor2** [comment]
Have you set single process?

>Application error mypal.exe, version 68.12.0.8721, module xul.dll, version 68.12.0.8721,

This is useless, you may give the dr watson log

**Comment by etsenberg** [comment]
> Have you set single process?

What is it?

> This is useless, you may give the dr watson log

[drwatson.txt](https://github.com/user-attachments/files/16034388/drwatson.txt)
Can I do anything else before rebooting?

**Comment by Feodor2** [comment]
I got to this finally
Please elaborate how to hang the browser like this, may be a specific site, whatever
Then tell what exact task do you kill
Also what about other browsers: firefox, roytam's and else

**Comment by etsenberg** [comment]
> I got to this finally Please elaborate how to hang the browser like this, may be a specific site, whatever Then tell what exact task do you kill Also what about other browsers: firefox, roytam's and else

As O've said, I have absolutely no idea why and when it happens. It doesn't look like a bug related to a specific webpage. It always happens after Mypal was not an active task - when switching back from another task or starting after a normal exit. And then in fails to display its window regardless of any closing and restarting until the system is rebooted, I have never seen this in any other browser - they can crash, but they always can be normally restarted, And I saw it in all versions of Mypal68 I used - probably starting from 7.

**Comment by Feodor2** [comment]
I asked you to tell what you do exact when browser hung, actually about "Killing and restarting"

If this not possible to reproduce, then I can give only tips.
Take clean install, look for software and drivers might interfere.

**Comment by etsenberg** [comment]
> I asked you to tell what you do exact when browser hung, actually about "Killing and restarting"
> 
> If this not possible to reproduce, then I can give only tips. Take clean install, look for software and drivers might interfere.

I close the non-working mypal by Alf-F4 or by killing all mypal.exe processes in the task manager - there is no difference in the result. BTW, task manager does not show mypal as "not responding" - it thinks it runs normally. I restart it the usual way - by clicking the icon I always click, which points to mypal.exe. I also tried different versions from different folders by running their mypal.exe from there - the result is the same: after the bug happens none starts normally The system shows mypal in the taskbar, but when switching to it (by clicking or by Alt-Tab), nothing is displayed (only the border of mypal window can be seen if it is not maximized, but only the border - the rectangle w/o any interface elements inside. Inside this border, there is only the picture which remained from the previous active task, which remains unchanged when the actual picture of that task under the mypal window changes). So, it looks as if mypal (and the system) thinks that it works, but it cannot display anything - even the menu bars (nor even the minimize and close buttons in the right corner), Switching from maximized to window mode and back or changing the display resolution do not help.
There are no unusual tasks which can interfere with the browser. I tried closing everything as the last step before rebooting, and still it did not help.

**Comment by Ariel-Schnee** [comment]
Maybe you should use 69.13.7b (32-bit) instead. I'm using it. And no crashes.^_^



---

Signature - SFW - https://www.fanficparadise.com/arielschnee/sig3.htm
Signature - NSFW - https://www.fanficparadise.com/arielschnee/sig3_NSFW.htm


  ----- Original Message -----

**Comment by RamonUnch** [comment]
> I close the non-working mypal by Alf-F4 or by killing all mypal.exe processes in the task manager - there is no difference in the result. BTW, task manager does not show mypal as "not responding" - it thinks it runs normally.

sounds like some buggy drivers that are unable to release resources even when the browser is terminated, I would suggest to completely disable hardware acceleration on MyPal and see if it still happens.

**Comment by etsenberg** [comment]
> sounds like some buggy drivers that are unable to release resources even when the browser is terminated, I would suggest to completely disable hardware acceleration on MyPal and see if it still happens.

I wrote it above. Disabled.

**Comment by Feodor2** [comment]
So does it really close or killed, you check in the advanced program (process hacker), anyway I ask to get it for further checks
And did you take clean install?

**Comment by etsenberg** [comment]
> So does it really close or killed, you check in the advanced program (process hacker), anyway I ask to get it for further checks And did you take clean install?

I don't understand what you are asking for. I've said, all mypal.exe processes were killed (disappeared from the process list), and still something prevented normal restarting, Right now everything works because I rebooted the system several days ago, and I don't know when the bug happens again. But it will. If you have any debugging or logging tool which I can install and wait for the bug to get more info, I am ready to install it (if it works in Win32 XP).
I don't know what you mean by "clean install". As there is no setup, each version was just unpacked from its archive into its own folder and run from it. I don't know what can be cleaner. And when the bug happens, none of mypal versions works, not only the one which caused the bug.

**Comment by Feodor2** [comment]
Get "process hacker" tool and check with it.
"clean instal" means that the profile also to be new clean one, assume you know what is the profile, if don't then learn what is "firefox profile"

**Comment by etsenberg** [comment]
> Get "process hacker" tool and check with it.

Isn't Sysinternals Suite procexp enough for you?
![processes](https://github.com/Feodor2/Mypal68/assets/159142903/acae773f-cf09-4ce9-806d-a7e7f0a76d47)
As I've said, even closing all tasks, leaving only system ones, does not fix the bug. 

> "clean instal" means that the profile also to be new clean one, assume you know what is the profile, if don't then learn what is "firefox profile"

Running my main browser w/o my profile is of no use for me. I could do it for a day, but not for months, waiting for the bug,

**Comment by Feodor2** [comment]
Whatever then, you may do one more thing beside mentioned above: when it hung next time make minidump with your tool of every mypal process and send them here.

**Comment by etsenberg** [comment]
> Whatever then, you may do one more thing beside mentioned above: when it hung next time make minidump with your tool of every mypal process and send them here.

Did you check drwatson log which I posted above? I think it contains all possible dumps,
As for this tool, this is the info it displays about a process (currently all works normally):
![image](https://github.com/Feodor2/Mypal68/assets/159142903/ca6dcdee-305c-4670-a87e-6696bd5f4004)
Is it helpful?

**Comment by etsenberg** [comment]
BTW, is there any command line switch which allows starting Mypal in safe mode when GUI does not work? If no, adding it to the next release would be useful.

**Comment by Ariel-Schnee** [comment]
Clean install is...

C:\Documents and Settings\???\Application Data\Mypal68

Delete Mypal68 folder.

Then C:\Documents and Settings\???\Local Settings\Application Data\Mypal68

Delete Mypal68 folder.



---

Signature - SFW - https://www.fanficparadise.com/arielschnee/sig3.htm
Signature - NSFW - https://www.fanficparadise.com/arielschnee/sig3_NSFW.htm


  ----- Original Message -----

**Comment by etsenberg** [comment]
> Clean install is... C:\Documents and Settings\???\Application Data\Mypal68 Delete Mypal68 folder. Then C:\Documents and Settings\???\Local Settings\Application Data\Mypal68 Delete Mypal68 folder.

Are you crazy to advise such things?! Luckily I am not ignorant enough to do that, but someone may be and will lose all his data! You should say "rename" or "move", not "delete"!

**Comment by Ariel-Schnee** [comment]
I thought everyone would have common sense enough to back them up first before deleting.

Oh well, fine. Here. Modified the instructions.

Clean install is...

C:\Documents and Settings\???\Application Data\Mypal68

Backup, then delete Mypal68 folder.

C:\Documents and Settings\???\Local Settings\Application Data\Mypal68

Backup, then delete Mypal68 folder.



---

Signature - SFW - https://www.fanficparadise.com/arielschnee/sig3.htm
Signature - NSFW - https://www.fanficparadise.com/arielschnee/sig3_NSFW.htm


  ----- Original Message -----

**Comment by Feodor2** [comment]
You drwtson about the crash when restarting, I want to look what happened before.
Okey just show something like this when browser hung
![image](https://github.com/Feodor2/Mypal68/assets/19492771/804ee77f-f127-4281-8888-6af08c54a110)
And what about other programs, it may be intrusive you run  - remove and try.
![image](https://github.com/Feodor2/Mypal68/assets/19492771/b4de886c-6ab9-4f26-afc9-56b404983eb9)

Safe mode
![image](https://github.com/Feodor2/Mypal68/assets/19492771/7c3aa328-4e1d-442c-b314-ae326e7ef68f)

**Comment by etsenberg** [comment]
> You drwtson about the crash when restarting, I want to look what happened before. 

No problem, here is the full log starting from March 25, 2024 (which also contains some previous mypal crashes)
[drwatsonlog.txt](https://github.com/user-attachments/files/16120499/drwatsonlog.txt)

Nv*.* processes are part of Nvidia drivers.
stunnel is a proxy which was NOT installed at the time of crash (see the after-the-crasg screenshot, not the current one)
apctray is a tray icon for UPS, and unblockassistant helps to unblock and delete shared files. I never used it to delete any mypal files, and I didn't use it at all for many days before the crash, so it cannot be the reason.

**Comment by Feodor2** [comment]
I did not asked another drwatsonlog.txt here.
So you refuse to do things I asked, then no more clues left, I withdraw this.

**Comment by etsenberg** [comment]
> I did not asked another drwatsonlog.txt here. So you refuse to do things I asked, then no more clues left, I withdraw this.

Are you enjoying mocking your users?!
I did what you said, If you mean something opposite to what you say, I am not a telepath!

> You drwtson about the crash when restarting, I want to look what happened before.

So I gave you the log which includes what happened before!

> Okey just show something like this when browser hung

I cannot do it before the browser hangs again!

> And what about other programs, it may be intrusive you run - remove and try.

I told you what these programs are and why they are not related to the browser, And again, I cannot try anything until the bug happens again!
And BTW, about safe mode. To activate it from abort:support, the browser GUI must work, When the bug is active, it is not possible. That's why I asked for something like "mypal.exe /safe" command line switch.

**Comment by Feodor2** [comment]
Just do things exactly what I asked.
commandline switch is --safe-mode

**Comment by NS-Clone** [comment]
> It happens after days or weeks of normal work w/o rebooting. I don't know the specific circumstances. Just switching to another task, then switching back - and it hangs. Killing and restarting does not help (the task starts and remains in memory, but nothing is displayed in its window), hibernation and restoring does not help - only rebooting the system. It happened in previous versions and happens in 68.14. When it happens, ALL Mypal versions stop working the same way - none can run until reboot, even if they use different profiles. After rebooting all works again w/o additional actions, so it's not in the profile - it must be something in the memory which does not disappear even when all mypal.exe in the process manager are killed. All other tasks work normally. There is a lot of free memory. Right now I have this situation. What can I do to diagnose the issue? I will not reboot, waiting for your answer.

what about your sound card?
with usb audio and bad usb connection
you can get some "wrong state"
at which some software doesn't runs at all even
(when they measuring time by sound)
with this glithces mypal works moar unstable and often crashing
(also looks like without sound card mypal works moar unstable too)

**Comment by etsenberg** [comment]
I don't have a separate sound card, it's integrated in the motherboard.

**Comment by newbie-461** [comment]
https://github.com/Feodor2/Mypal68/issues/235

**Comment by etsenberg** [comment]
It happened again (53 days passed since the previous time). I restarted already. but before doing it I tried to run mypol in safe mode from command prompt, and that's what I saw:
![063](https://github.com/user-attachments/assets/bed3c93a-ecd7-4caf-a817-420f11cb9568)
(the gray rectangle was actually transparent, displaying the desktop)
As I was unable to click anything in the empty window, I pressed Enter, and it did not help (mypal tried to run, but the bug was back). After rebooting the system, it started in normal mode and restored all tabs correctly w/o asking,
So, when the bug happens, mypal fails to display the content of ANY window, be it its main window or dialog window.
Here is the list of the processes which were running:
![064](https://github.com/user-attachments/assets/2002556e-3fdd-461e-92be-a304ab518f71)

**Comment by etsenberg** [comment]
Update. I said that it happened after 53 days, but at the previous time I did not reboot for several days after the bug. So looks like it indeed happens on 49th day, as it was mentioned [above](https://github.com/Feodor2/Mypal68/issues/235),

**Comment by Feodor2** [comment]
The bug is well-known, and this is almost unrealistic scenario, never I did not tried to fix, and probably never would, only few user runs browser for so long.  Assume this is duplicate

**Comment by etsenberg** [comment]
> The bug is well-known, and this is almost unrealistic scenario, never I did not tried to fix, and probably never would, only few user runs browser for so long. Assume this is duplicate

It's not running the browser because I definitely closed and restarted it during this period several times due to memory leaks, It's running the computer w/o rebooting (but the bug happens ONLY in mypal!)

**Comment by newbie-461** [comment]
[mozglue.zip](https://github.com/user-attachments/files/16858874/mozglue.zip)

i made some dirty hacks, apply by replacing mozglue.dll
it will get stuck at every ~25th day of uptime, but recover after reopening Mypal68, system reboot is not needed

**Comment by etsenberg** [comment]
> [mozglue.zip](https://github.com/user-attachments/files/16858874/mozglue.zip)
> 
> i made some dirty hacks, apply by replacing mozglue.dll it will get stuck at every ~25th day of uptime, but recover after reopening Mypal68, system reboot is not needed

So, do you understand the reason of the bug?

**Comment by newbie-461** [comment]
> So, do you understand the reason of the bug?

Not sure, but I think the bug is around [CompositorVsyncScheduler.cpp](https://github.com/Feodor2/Mypal68/blob/main/gfx/layers/ipc/CompositorVsyncScheduler.cpp)
I will make a PR if i find out the reason/solution.

**Comment by etsenberg** [comment]
> [mozglue.zip](https://github.com/user-attachments/files/16858874/mozglue.zip)
> 
> i made some dirty hacks, apply by replacing mozglue.dll it will get stuck at every ~25th day of uptime, but recover after reopening Mypal68, system reboot is not needed

Now I can confirm that it works! Thank you for this solution. Too bad that [Feodor2] doesn't want to fix this bug :(

**Comment by etsenberg** [comment]
[mozglue.zip](https://github.com/user-attachments/files/16858874/mozglue.zip)

[newbie-461](https://github.com/newbie-461), can you make the same thing for ver. 14.7b? The old one doesn't work there.
